osu.Game.Rulesets.Mania/UI/ManiaPlayfieldAdjustmentContainer.cs

```diff
@@ -1,17 +1,64 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
+using osu.Framework.Allocation;
 using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
 using osu.Game.Rulesets.UI;
+using osuTK;
 
 namespace osu.Game.Rulesets.Mania.UI
 {
     public partial class ManiaPlayfieldAdjustmentContainer : PlayfieldAdjustmentContainer
     {
+        protected override Container<Drawable> Content { get; }
+
+        private readonly DrawSizePreservingFillContainer scalingContainer;
+
         public ManiaPlayfieldAdjustmentContainer()
         {
-            Anchor = Anchor.Centre;
-            Origin = Anchor.Centre;
+            InternalChild = scalingContainer = new DrawSizePreservingFillContainer
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                RelativeSizeAxes = Axes.Both,
+                Child = Content = new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                }
+            };
+        }
+
+        [Resolved]
+        private DrawableManiaRuleset drawableManiaRuleset { get; set; } = null!;
+
+        protected override void Update()
+        {
+            base.Update();
+
+            float aspectRatio = DrawWidth / DrawHeight;
+            bool isPortrait = aspectRatio < 4 / 3f;
```

Comment:
Should use `osu.Framework.Utils.Precision`, or just compare with `1f`. Otherwise a 4:3 tablet in landscape might be considered portrait by float imprecision.

Another reason to use `1f` is that older 1280x1024 displays (5:4) are properly considered as landscape.